src/browser.ts

```diff
@@ -120,11 +124,34 @@ export class GrpcClient {
     configOverrides: gax.ClientConfig,
     headers: OutgoingHttpHeaders
   ) {
+    function buildMetadata(abTests, moreHeaders) {
+      const metadata = {};
+      if (moreHeaders) {
+        for (const key in moreHeaders) {
+          if (
+            key.toLowerCase() !== 'x-goog-api-client' &&
+            moreHeaders.hasOwnProperty(key)
+          ) {
+            const value = moreHeaders[key];
+            if (Array.isArray(value)) {
+              value.forEach(v => {
+                metadata[key] = v;
```

Comment:
I could be interpreting this in  correctly but doesn't this code just iterate over all of  the items  in `value`  and overwrite what was in `metadata[key]` with `v`? Shouldn't we be appending or copying the array?